homeassistant/components/fritz/common.py

```diff
@@ -386,64 +379,20 @@ async def async_trigger_firmware_update(self) -> bool:
 
     async def async_trigger_reboot(self) -> None:
         """Trigger device reboot."""
-        await self.hass.async_add_executor_job(
-            self.connection.call_action, "DeviceConfig1", "Reboot"
-        )
+        await self.connection.reboot()
 
     async def async_trigger_reconnect(self) -> None:
         """Trigger device reconnect."""
-        await self.hass.async_add_executor_job(
-            self.connection.call_action, "WANIPConn1", "ForceTermination"
-        )
-
-    async def service_fritzbox(
-        self, service_call: ServiceCall, config_entry: ConfigEntry
-    ) -> None:
-        """Define FRITZ!Box services."""
-        _LOGGER.debug("FRITZ!Box service: %s", service_call.service)
-
-        if not self.connection:
-            raise HomeAssistantError("Unable to establish a connection")
-
-        try:
-            if service_call.service == SERVICE_REBOOT:
-                _LOGGER.warning(
-                    'Service "fritz.reboot" is deprecated, please use the corresponding button entity instead'
-                )
-                await self.hass.async_add_executor_job(
-                    self.connection.call_action, "DeviceConfig1", "Reboot"
-                )
-                return
+        await self.connection.reconnect()
```

Comment:
this needs to be wrapped into `hass.async_add_executor_job()`

Reply:
Fixed